SourceX/display.cpp

```diff
@@ -17,6 +17,116 @@
 #endif
 #endif
 
+namespace {
+
+int screenWidth = -1;
+int screenHeight = -1;
+
+int viewportHeight = -1;
+
+void ForceScreenResInit()
+{
+	bool videoInitialized = SDL_WasInit(SDL_INIT_VIDEO);
+
+	if (!videoInitialized) {
+		SDL_Init(SDL_INIT_VIDEO);
+	}
+	disp::InitDesiredScreenRes();
+	if (!videoInitialized) {
+		SDL_Quit();
+	}
+}
+
+void ForceViewportHeightInit()
+{
+	if (screenWidth == -1) {
+		ForceScreenResInit();
+	}
+
+	disp::InitViewportHeight();
+}
+
+} // anonymous namespace
+
+namespace disp {
+
+void InitDisplayElementSizes()
+{
+	if (screenWidth == -1) {
+		InitDesiredScreenRes();
+	}
+
+	if (viewportHeight == -1) {
+		InitViewportHeight();
+	}
+}
+
+void InitDesiredScreenRes()
+{
+#ifdef USE_SDL1
+	// Use the default 640x480 resolution when attempting to compile with SDL 1.2.
+	screenWidth = 640;
+	screenHeight = 480;
+#else
+	SDL_DisplayMode mode;
+
+	if (SDL_GetDesktopDisplayMode(0, &mode) != 0) {
+		ErrSdl();
+	}
+
+	screenWidth = mode.w;
+	dvl::DvlIntSetting("screen width", &screenWidth);

```

Comment:
Instead of going for the max possible resolution by default, I have adjusted things so that it will adjust the configured resolution to the aspect of the screen. Meaning that by default we will simply remove the letterboxing. But this also gave me some good ideas for how to integrate it with integer scaling.

To disable this behavior I have added another option `original aspect ratio`. You can still set the game to run at your monitor resolution, but you will now have to set it in the ini, instead of the other way around.